### PR TITLE
Adding option to use Giphy as a source for animated images.

### PR DESCRIPTION
**Pull request by liquidmetal**
To use this patch, add a new environment variable called HUBOT_USE_GIPHY and set its value to "true". If you use "animate me" and the env var is set, giphy will be used. If you use "image me" google will be used whether or not HUBOT_USE_GIPHY is set.

Docs for the Giphy API: https://github.com/giphy/GiphyAPI#search-endpoint

### CODE REVIEW

**Comment by HaroldPutman** [comment]
Thanks for adding this!
* Rather than a boolean, we should enable this by defining an API key like `HUBOT_GIPHY_KEY=`. We could provide the beta key in the setup instructions for people who don't want to bother requesting a real one, but we should not hardcode that. There may be rate limits.
* The [Giphy terms of service](http://giphy.com/terms) require us to show "Powered by Giphy". so we should send that text before or after the image whenever we use Giphy.
* I'd like to see us put `rating: "g"` in the request. This could be configurable, but I don't want hubot to get fired in more restrictive corporate environments ;-)

**Comment by liquidmetal** [comment]
* Makes sense. I've added `HUBOT_GIPHY_KEY`
* I've put in a `(fetched from Giphy)` message after the gif
* I've set `g` to be the default rating. You can override this by using `HUBOT_GIPHY_RATING`

**Comment by thatarchguy** [comment]
+1! Running on my instance at work right now.

**Comment by HaroldPutman** [comment]
Can we change "(fetched from Giphy)" to "Powered by Giphy". They are pretty specific in the terms about that text. Otherwise this looks fantastic.

File: src/google-images.coffee [hunk 1]
```diff
@@ -83,7 +85,27 @@ imageMe = (msg, query, animated, faces, cb) ->
...
-    msg.http('https://ajax.googleapis.com/ajax/services/search/images')
+    url = 'https://ajax.googleapis.com/ajax/services/search/images'
+    use_giphy = false
+
+    if typeof animated is 'boolean' and animated is true
+      giphy_key = (process.env.HUBOT_GIPHY_KEY)
+      use_giphy = (typeof giphy_key is 'string' and giphy_key.length > 0)
+      if typeof use_giphy is 'boolean' and use_giphy is true
+        url = 'http://api.giphy.com/v1/gifs/search'
+        rating = process.env.HUBOT_GIPHY_RATING
```
Comment (HaroldPutman) [inline src/google-images.coffee]: Since movie ratings are usually written like "G" and "PG" lets add `.toLowerCase()` here to avoid that likely mistake.

**Comment by HaroldPutman** [comment]
There have been some objections to using Giphy as a sole source of animated Gifs. (see discussion in issue #13), plus there have been some new developments in getting the google image search to work. 
It seems more fitting to stick to the google results while we can., so I am going to close this PR in favor of the approach in PR #15 . Thanks for your work on this.

**Pull request closed without merging**